src/hooks/useActiveProject.tsx

```diff
@@ -50,35 +59,54 @@ export const ActiveProjectContextProvider = ({
         return;
       }
 
-      // TODO: Save for previously-selected project
+      storeProjectId(selectedProject.id);
 
       setHookReturnValue({
         activeProject: selectedProject,
         activeSubjectId: selectedSubject.subjectId,
       });
     },
-    [projectsResult.data, useMeResult.data],
+    [projectsResult.data, useMeResult.data, storeProjectId],
   );
 
   useEffect(() => {
-    if (projectsResult.data?.length) {
-      // TODO: Load previously-selected project
-
-      const firstProjectId = projectsResult.data?.[0].id;
-      if (firstProjectId) {
-        setActiveProjectId(firstProjectId);
+    if (projectsResult.data?.length || storedProjectIdResult.isFetched) {
+      const storedProjectId = projectsResult.data?.find(
+        (project) => project.id === storedProjectIdResult.data,
+      )?.id;
+      const selectedProjectId = storedProjectId ?? projectsResult.data?.[0].id;
+      if (selectedProjectId) {
+        setActiveProjectId(selectedProjectId);
+        storeProjectId(selectedProjectId);
       }
     }
-  }, [projectsResult.data, setActiveProjectId]);
+  }, [
+    projectsResult.data,
+    setActiveProjectId,
+    storedProjectIdResult.isFetched,
+    storedProjectIdResult.data,
+    storeProjectId,
+  ]);
 
   return (
     <ActiveProjectContext.Provider
       value={{
         ...hookReturnValue,
         setActiveProjectId,
-        isLoading: !!(projectsResult.isLoading || useMeResult.isLoading),
-        isFetched: !!(projectsResult.isFetched && useMeResult.isFetched),
-        error: projectsResult.error || useMeResult.error,
+        isLoading: !!(
+          projectsResult.isLoading ||
+          useMeResult.isLoading ||
+          storedProjectIdResult.isLoading
+        ),
+        isFetched: !!(
+          projectsResult.isFetched &&
+          useMeResult.isFetched &&
+          storedProjectIdResult.isFetched
+        ),
+        error:
+          projectsResult.error ||
+          useMeResult.error ||
+          storedProjectIdResult.error,
```

Comment:
I'm not sure we'd necessarily want to return the `storedProjectIdResult.error` here.  It feels like an error that could be recovered from (e.g. the user just needs to re-select the project) whereas the others are errors consumers will need to handle.

Reply:
True enough, we could always just fall back to use the default in that event.